storage/shard.go

```diff
@@ -370,9 +485,9 @@ func (s *dbShard) FetchBlocks(
 	starts []time.Time,
 ) []block.FetchBlockResult {
 	s.RLock()
-	entry, _, exists := s.getEntryWithLock(id)
+	entry, _, err := s.lookupEntryWithLock(id)
 	s.RUnlock()
-	if !exists {
+	if err != nil {
```

Comment:
Do we not want to return the actual error here if `err` is not `errShardEntryNotFound`?